File: Sources/AdbShell/Models/FDroidUpdateInfo.swift
import Foundation

/// Найденное на F-Droid обновление для установленного пакета.
struct FDroidUpdateInfo: Identifiable, Equatable {
    var id: String { packageName }
    let packageName: String
    let installedVersionCode: Int
    let latestVersionCode: Int
    let latestVersionName: String?

    /// Официальный URL прямой сборки F-Droid: repo/<pkg>_<versionCode>.apk.
    var downloadURL: URL {
        URL(string: "https://f-droid.org/repo/\(packageName)_\(latestVersionCode).apk")!
    }

    var fdroidPageURL: URL {
        URL(string: "https://f-droid.org/packages/\(packageName)/")!
    }
}

File: desktop/src/main/adb/types/FDroidUpdateInfo.ts
// Порт Sources/AdbShell/Models/FDroidUpdateInfo.swift

export interface FDroidUpdateInfo {
  packageName: string;
  installedVersionCode: number;
  latestVersionCode: number;
  latestVersionName?: string;
}

/** Официальный URL прямой сборки F-Droid: repo/<pkg>_<versionCode>.apk. */
export function fdroidDownloadUrl(info: FDroidUpdateInfo): string {
  return `https://f-droid.org/repo/${info.packageName}_${info.latestVersionCode}.apk`;
}

export function fdroidPageUrl(info: FDroidUpdateInfo): string {
  return `https://f-droid.org/packages/${info.packageName}/`;
}
